crates/pathfinder/src/state/sync.rs

```diff
@@ -302,7 +353,36 @@ where
 
                     tracing::trace!("Query for existence of contracts: {:?}", contracts);
                 }
+                Some(l2::Event::Pending(block, diff)) => {
+                    // Update state tree to determine new state root, but rollback the changes as we do
+                    // not want to persist them.
+                    let new_root = tokio::task::block_in_place(|| {
+
+                        let tx = db_conn.transaction()?;
```

Comment:
Since `update_starknet_state` _will_ try to write into the DB don't we need `db_conn.transaction_with_behavior(TransactionBehavior::Immediate)` here (to avoid the transaction upgrade to write)?

Reply:
They already are I think, see PR #454